controller/db/interface_db_model.go

```diff
@@ -0,0 +1,137 @@
+	MTU             int64    `json:"mtu"`
+	Index           int64    `json:"index"`
+	Flags           uint64   `json:"flags"`
+	Addresses       []string `json:"addresses"`
+}
+
+func (self *Interface) IsUp() bool {
+	return self.IsFlagSet(net.FlagUp)
+}
+
+func (self *Interface) IsRunning() bool {
+	return self.IsFlagSet(net.FlagRunning)
+}
+
+func (self *Interface) IsLoopback() bool {
+	return self.IsFlagSet(net.FlagLoopback)
+}
+
+func (self *Interface) IsBroadcast() bool {
+	return self.IsFlagSet(net.FlagBroadcast)
+}
+
+func (self *Interface) IsMulticast() bool {
+	return self.IsFlagSet(net.FlagMulticast)
+}
+
+func (self *Interface) IsFlagSet(f net.Flags) bool {
+	return net.Flags(self.Flags)&f == f
+}
+
+func (self *Interface) MarshalJSON() ([]byte, error) {
+	result := map[string]any{}
+	result["name"] = self.Name
+	result["hardwareAddress"] = self.HardwareAddress
+	result["mtu"] = self.MTU
+	result["index"] = self.Index
+	result["isBroadcast"] = self.IsBroadcast()
+	result["isLoopback"] = self.IsLoopback()
+	result["isMulticast"] = self.IsMulticast()
+	result["isRunning"] = self.IsRunning()
+	result["isUp"] = self.IsUp()
+	result["addresses"] = self.Addresses
+
+	return json.Marshal(result)
+}
+
+func (self *Interface) toBoltMap() map[string]any {
+	result := map[string]any{}
+	result[FieldName] = self.Name
+	result[FieldInterfaceHardwareAddress] = self.HardwareAddress
+	result[FieldInterfaceMtu] = self.MTU
+	result[FieldInterfaceIndex] = self.Index
+	result[FieldInterfaceFlags] = int64(self.Flags)
+
+	addresses := make([]any, len(self.Addresses))
+	for i, address := range self.Addresses {
+		addresses[i] = address
+	}
+	result[FieldInterfaceAddresses] = addresses
+	return result
+}
+
+func (self *Interface) FillEntity(bucket *boltz.TypedBucket) {
+	self.Name = bucket.GetStringOrError(FieldName)
+	self.HardwareAddress = bucket.GetStringOrError(FieldInterfaceHardwareAddress)
+	self.MTU = bucket.GetInt64WithDefault(FieldInterfaceMtu, 0)
+	self.Index = bucket.GetInt64WithDefault(FieldInterfaceIndex, 0)
+	self.Flags = uint64(bucket.GetInt64WithDefault(FieldInterfaceFlags, 0))
+	addresses := bucket.GetList(FieldInterfaceAddresses)
+	for _, address := range addresses {
+		self.Addresses = append(self.Addresses, address.(string))
+	}
+}
+
+func loadInterfaces(bucket *boltz.TypedBucket) []*Interface {
+	var result []*Interface
+	if interfacesBucket := bucket.GetBucket(FieldInterfaces); interfacesBucket != nil {
+		err := interfacesBucket.ForEachTypedBucket(func(_ string, intfBucket *boltz.TypedBucket) error {
+			intf := &Interface{}
+			intf.FillEntity(intfBucket)
+			result = append(result, intf)
+			return bucket.Err
+		})
+
+		if err != nil {
+			bucket.SetError(err)
+		}
+	}
+	return result
+}
+
+func storeInterfaces(interfaces []*Interface, ctx *boltz.PersistContext) {
+	if ctx.ProceedWithSet(FieldInterfaces) {
+		m := map[string]any{}
+		for idx, v := range interfaces {
+			m[fmt.Sprintf("%d.%s", idx, v.Name)] = v.toBoltMap()
```

Comment:
So this uses the report order + interface name as the unique key? If the OS reports the order differently, it would have to be an entirely new map with every update. 

It isn't an issue as long as we never directly reference the key values as foreign keys and only use this as information for a user to make other decisions. Otherwise, MAC address would probably have been better.